File: Yow.CoD.Finance.NancyWebHost/Models/CreateLoanModel.cs
using System;
using Yow.CoD.Finance.Domain.Contracts;

namespace Yow.CoD.Finance.NancyWebHost.Models
{
    public sealed class CreateLoanModel
    {
        public string CustomerName { get; set; }
        public string PreferredPhoneNumber { get; set; }
        public string AlternatePhoneNumber { get; set; }
        public string PostalAddress { get; set; }
        public string BankBsb { get; set; }
        public string BankAccount { get; set; }
        public decimal LoanAmount { get; set; }

        public CreateLoanCommand ToCommand()
        {
            var customerContact = new CustomerContact(CustomerName, PreferredPhoneNumber, AlternatePhoneNumber,
                PostalAddress);
            var bankAccount = new BankAccount(BankBsb, BankAccount);
            var duration = new Duration(12, DurationUnit.Month);
            var command = new CreateLoanCommand(Guid.NewGuid(), Guid.NewGuid(), DateTimeOffset.Now,
                customerContact,
                bankAccount,
                PaymentPlan.Weekly,
                LoanAmount,
                duration);
            return command;
        }
    }
}

File: java/web-comms-adapter/src/main/java/com/leecampbell/cod/web/CreateLoanModel.java
package com.leecampbell.cod.web;

import com.leecampbell.cod.domain.contracts.*;
import java.math.BigDecimal;
import java.time.OffsetDateTime;
import java.util.UUID;

public final class CreateLoanModel {
    public String customerName;
    public String preferredPhoneNumber;
    public String alternatePhoneNumber;
    public String postalAddress;
    public String bankBsb;
    public String bankAccount;
    public BigDecimal loanAmount;

    public CreateLoanCommand toCommand() {
        CustomerContact customerContact = new CustomerContact(customerName, preferredPhoneNumber, alternatePhoneNumber,
                postalAddress);
        BankAccount account = new BankAccount(bankBsb, bankAccount);
        Duration duration = new Duration(12, DurationUnit.Month);
        return new CreateLoanCommand(UUID.randomUUID(), UUID.randomUUID(), OffsetDateTime.now(),
                customerContact, account, PaymentPlan.Weekly, loanAmount, duration);
    }
}
